### DISCUSSION TITLE
Debug logs on devnet

### DISCUSSION THREAD

**Discussion by lumos42** [aptos-labs/aptos-developer-discussions#348] [General]
Hi, Is there a way to view debug logs on devnet? By debug logs, I mean logs using `std::debug::print`

**Comment by 0xjunha** [comment]
You can get debugging logs in the following ways:

1. Run Move unit tests (`aptos move test`)
2. Simulate transaction locally (`aptos move run` for running entry functions or `aptos move run-script` running scripts with `--local` flag)
3. Replay transactions that have already executed on mainnet/testnet/devnet (`aptos move replay`)

For a full execution trace of a transaction on mainnet/testnet/devnet, you can use transaction replay with `--profile-gas` flag. This will provide a complete execution trace in the gas profiling report.

**Reply by 0xshinobii** [reply]
this works. thanks
